### PR TITLE
Do not assume the "public" schema

### LINKED ISSUE #230: Spatial Index assuming public schema
When creating a spatial index for a declaratively defined table that does not specify the schema, `public` is assumed to be the schema for the table (https://github.com/geoalchemy/geoalchemy2/blob/master/geoalchemy2/__init__.py#L81). My issue here is I have the default schema for my DB set up to be `my_schema`, so my table is created in that schema, then when alembic tries to generate the spatial index for the column geoalchemy is prepending a `public.` to the index table, thereby causing the index creation to fail.

### PR DESCRIPTION
**Pull request by elemoine**
This PR aims to fix #230, where GeoAlchemy assumes that the schema is "public" when no schema is specified by the user.

Closes #230.

### CODE REVIEW

**Comment by elemoine** [comment]
WIP for the moment. Needs testing, and unit-tests probably.

**Comment by elemoine** [comment]
Now with tests…

File: geoalchemy2/__init__.py [hunk 1]
```diff
@@ -105,10 +103,15 @@ def dispatch(event, table, bind):
...
-                                      table.name, c.name))
+                        if table.schema:
+                            bind.execute('CREATE INDEX "idx_%s_%s" ON "%s"."%s" '
+                                         'USING GIST ("%s")' %
+                                         (table.name, c.name, table.schema,
+                                          table.name, c.name))
+                        else:
+                            bind.execute('CREATE INDEX "idx_%s_%s" ON "%s" '
+                                         'USING GIST ("%s")' %
+                                         (table.name, c.name, table.name, c.name))
```
Comment (loicgasser) [inline geoalchemy2/__init__.py]: What about a helper function `table_ref(table)`  that returns the string? You could avoid the `if/else` and the duplication of SQL.

Comment (elemoine) [reply geoalchemy2/__init__.py]: At the same time the current code is very easy to read and understand. So I tend to think that the duplication is ok here.

Comment (elemoine) [reply geoalchemy2/__init__.py]: Loïc, I am going to merge this. But feel free to open a new PR if you think you can improve the code. I'll be happy to merge any improvement. Thanks.

Comment (loicgasser) [reply geoalchemy2/__init__.py]: I was just passing by, it was a very superficial look at it.

**Pull request merged**